lib/parsers/iso_base_media_file_format/decoder.rb

```diff
@@ -1017,22 +1041,16 @@ def uuid(size)
       #
       # See https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/QTFFChap4/qtff4.html#//apple_ref/doc/uid/TP40000939-CH206-18737.
       def read_matrix
-        9.times.map do |i|
-          if i % 3 == 2
-            read_fixed_point_32_2_30
-          else
-            read_fixed_point_32
-          end
-        end
+        Matrix.build(3) { |_, c| read_fixed_point(fractional_digits: c % 3 == 2 ? 30 : 16, signed: true) }
```

Comment:
Curious, does `matrix` significantly improve this logic, as opposed to the more rudimentary approach we currently have?

Reply:
_Significantly improve?_ Probably not, but feel like it represents the intentions of the matrix fields better than a flat 9-element array, and the value is then immediately ready for use in any calculations.